src/player/player.service.js

```diff
@@ -932,7 +932,7 @@ angular.module('ngGo.Player.Service', [
 
           //Call listener
           listener.apply(this, arguments);
-        });
+        }.bind(this));
```

Comment:
Ok I thought that using arrow functions might blow up somewhere with the $scope soup we have.
Rather than binding though, let's make a reference to `this` above, e.g. `const self = this;` and use that. I'll make that change after merge.